### DISCUSSION TITLE
SQLModel cannot fetch the type of a enum field containing Annotated type

### DISCUSSION THREAD

**Discussion by fubuloubu** [fastapi/sqlmodel#903] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def iso_format(dt: datetime) -> str:
    return dt.isoformat()

ISOTimestamp = Annotated[
    datetime,
    PlainSerializer(iso_format, return_type=str),
]

class Product(BaseModel, table=True):
    created: ISOTimestamp = Field(default_factory=utc_now, nullable=False)
    deprecated: ISOTimestamp | None = Field(default=None)  # Works if I remove this field
```

### Description

When loading the above model I get
```
  ...
  File "site-packages/sqlmodel/main.py", line 474, in __new__
    col = get_column_from_field(v)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "site-packages/sqlmodel/main.py", line 618, in get_column_from_field
    sa_type = get_sqlalchemy_type(field)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "site-packages/sqlmodel/main.py", line 567, in get_sqlalchemy_type
    if issubclass(type_, Enum):
       ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: issubclass() arg 1 must be a class
```

### Operating System

Linux

### SQLModel Version

0.0.16

### Python Version

3.11.7

**Comment by joachimhuet** [comment, marked as answer]
In fact this error comes from sqlalchemy, you can see that the error disapeear if you remove the `tablle=True`

So what you can do is specify a `sa_type` to make sqlalchemy understand what type you actually want to be used in the database side.

Something like that should fix your problem:
```python
from sqlalchemy import DateTime  # if you want the DateTime type or String if you want it to be a string

class Product(BaseModel, table=True):
    created: ISOTimestamp = Field(default_factory=utc_now, nullable=False, sa_type=DateTime)
    deprecated: ISOTimestamp | None = Field(default=None, sa_type=DateTime)
```

**Reply by fubuloubu** [reply]
Ah okay, this makes sense as it skips the requirement to dynamically infer the SA type (just need to add nullable too)
